examples/comments.rs

```diff
@@ -57,13 +57,16 @@ impl Completer for CommentCompleter {
             // under the same condition, then
             // This condition is only false under the predicate that we are in a space with no
             // word to the left
+            let val: Option<FilenameCompleter>;
             if filename {
                 let completer = FilenameCompleter::new(Some(current_dir().unwrap()));
-                replace(&mut self.inner, Some(completer));
+                val = replace(&mut self.inner, Some(completer));
```

Comment:
Just some notes to discuss, am merging this PR.  I am not sure why they called replace like this instead of self.inner.replace- just seems unnecessarily confusing to me.  I assume you removed the unwraps because it seems they could cause an errant panic (ie if the previous inner was legit None) not because they are unwraps since an example like this is almost the textbook place to use unwrap to focus on the actual topic (but again it looks like they were a bug in this case).  Also, I do not think you need val just to make clippy happy, should be able to do something like "if self.inner.replace(..).is_none {}"- still useless boilerplate but no extra var.  Would probably be better to use 'take' and 'insert' in these places (or maybe not even mess with the completer...).

Finally, no idea why there is the same example with and without comments, that has always bugged me and we need to delete one and give the remaining one a proper name.  No idea why comments are bad especially for an example...